### DISCUSSION TITLE
How to use ClusterPipeline in production

### DISCUSSION THREAD

**Discussion by laoba25** [xetorthio/jedis#2767] [Q&A] [answered]
I am glad to hear that jedis finally supports the cluster pipeline.
Because of the lack of relevant documentation, we are exploring how to use it.

Does it need create ClusterConnectionProvider + ClusterPipeline object every time , and close after done ?

Example:

```java
    public static void test() {
        DefaultJedisClientConfig DEFAULT_CLIENT_CONFIG = DefaultJedisClientConfig.builder().build();

        Set<HostAndPort> CLUSTER_NODES = new LinkedHashSet<HostAndPort>();

        CLUSTER_NODES.add(new HostAndPort("127.0.0.1", 7000));

        ClusterConnectionProvider provider = new ClusterConnectionProvider(CLUSTER_NODES, DEFAULT_CLIENT_CONFIG);

        ClusterPipeline pipeline = new ClusterPipeline(provider);
        Response<String> resp=pipeline.set("K", "V");
        pipeline.sync();
        pipeline.close();
    }
```
Very Confused.
Really need some help，Thanks！

**Comment by sazzad16** [comment, marked as answer]
1. You can create ClusterConnectionProvider once and use in multiple non-parallel ClusterPipeline(s). The same provider can also be used multiple parallel ClusterPipeline(s) which would be limited by GenericObjectPoolConfig\<Connection\> of ClusterConnectionProvider.
2. You can directly create `new ClusterPipeline(CLUSTER_NODES, DEFAULT_CLIENT_CONFIG)`.

**Reply by laoba25** [reply]
Thanks! It's helpful!

**Reply by luxe** [reply]
But how do you turn a `JedisCluster` into a `ClusterPipeline`?  There is no corresponding `pipelined()` method.  There are no getters on JedisCluster which would give you the `CLUSTER_NODES` and `DEFAULT_CLIENT_CONFIG`.

**Reply by sazzad16** [reply]
> how do you turn a `JedisCluster` into a `ClusterPipeline`?

@luxe At this moment, there is no such option.
